packages/php/blueprint/src/Util.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint;
+
+use RecursiveArrayIterator;
+use RecursiveIteratorIterator;
+
+class Util {
+	public static function snake_to_camel( $string ) {
+		// Split the string by underscores
+		$words = explode( '_', $string );
+
+		// Capitalize the first letter of each word
+		$words = array_map( 'ucfirst', $words );
+
+		// Join the words back together
+		return implode( '', $words );
+	}
+
+	public static function array_flatten($array) {
+		return new RecursiveIteratorIterator(new RecursiveArrayIterator($array));
+	}
+
+	public static function ensure_wp_content_path($path) {
+		$path = realpath( $path );
+		if ( $path === false || strpos( $path, WP_CONTENT_DIR ) !== 0 ) {
+			throw new \InvalidArgumentException( "Invalid path: $path" );
+		}
+
+		return $path;
+	}
+
+	public static function camel_to_snake( $input ) {
+		// Replace all uppercase letters with an underscore followed by the lowercase version of the letter
+		$pattern     = '/([a-z])([A-Z])/';
+		$replacement = '$1_$2';
+		$snake       = preg_replace( $pattern, $replacement, $input );
+
+		// Replace spaces with underscores
+		$snake = str_replace( ' ', '_', $snake );
+
+		// Convert the entire string to lowercase
+		return strtolower( $snake );
+	}
+
+	public static function deactivate_plugin_by_slug( $slug ) {
+		if ( ! function_exists( 'deactivate_plugins' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/plugin.php';
+		}
+
+		// Get all installed plugins
+		$all_plugins = get_plugins();
+
+		// Loop through all plugins to find the one with the specified slug
+		foreach ( $all_plugins as $plugin_path => $plugin_info ) {
+			// Check if the plugin path contains the slug
+			if ( strpos( $plugin_path, $slug . '/' ) === 0 ) {
+				// Deactivate the plugin
+				deactivate_plugins( $plugin_path );
+			}
+		}
+		return false;
```

Comment:
Should we return `true` upon successful deactivation of a plugin?

Reply:
Also, just noticed that we have [UsePluginHelpers.php](https://github.com/woocommerce/woocommerce/pull/49763/files#diff-f19399f98627710f1add56629c587551f2ec37c93599b62deae71ff0ad34d85d), which is similar to these methods. Is it possible to simplify the code by reusing either of these methods?

Reply:
Good catch! These two functions in `Util.php` are no longer used. They are now in [UsePluginHelpers.php](https://github.com/woocommerce/woocommerce/pull/49763/files#diff-f19399f98627710f1add56629c587551f2ec37c93599b62deae71ff0ad34d85dR54)

The function in `UsePluginHelpers.php` correctly returns [true](https://github.com/woocommerce/woocommerce/pull/49763/files#diff-f19399f98627710f1add56629c587551f2ec37c93599b62deae71ff0ad34d85dR54)